ext/standard/password.c

```diff
@@ -61,7 +81,14 @@ static php_password_algo php_password_determine_algo(const char *hash, const siz
 {
 	if (len > 3 && hash[0] == '$' && hash[1] == '2' && hash[2] == 'y' && len == 60) {
 		return PHP_PASSWORD_BCRYPT;
+	} 
+#if HAVE_ARGON2LIB
+	if (hash[0] == '$' && strstr(hash, "argon2i")) {
```

Comment:
Because of the memory, time, and parallelism factors, Argon2 hashes can be of variable length. Using the default constants listed in `php_password.h`, this string will be 99 characters long, however the change of constants would result in different sized hashes.

The other alternative is to try to parse out all the available Argon2 options on the spot - there's still a chance that the string isn't an Argon2 hash though even verifying those options. At the end of the day the hash, will be password to `password_verify`. If the string isn't a valid Argon2 hash it'll be detected there and the `password_verify` will return false.

Thoughts?